### PR TITLE
Method for customizing typed persistent actor's recovery behavior

### PR DESCRIPTION
**Pull request by svroonland**
See #25216

### CODE REVIEW

**Comment by raboof** [comment]
OK TO TEST

**Comment by raboof** [comment]
(discussion on what functionality we want to expose here in the ticket at #25216)

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/PersistentBehaviors.scala [hunk 1]
```diff
@@ -98,6 +98,11 @@ trait PersistentBehavior[Command, Event, State] extends DeferredBehavior[Command
    */
   def onRecoveryCompleted(callback: (ActorContext[Command], State) ⇒ Unit): PersistentBehavior[Command, Event, State]
 
+  /**
+   * Change the recovery strategy
+   */
+  def withRecovery(recovery: Recovery): PersistentBehavior[Command, Event, State]
```
Comment (patriknw) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/PersistentBehaviors.scala]: I think we should define a new `Recovery` in `akka.persistence.typed`. We might not want to support all of the recovery strategies. For example I'm skeptical to `toSequenceNr` and `replayMax`.

Remove the `import akka.persistence._` in this file and replace with full class imports so we see the dependencies more clearly.

**Comment by patriknw** [comment]
Closing due to inactivity (and probably unclear directions from us)

**Pull request closed without merging**